### ISSUE TITLE
bzlmod integrity error messages are inaccurate and confusing

### ISSUE THREAD

**Issue by tyler-french** [bazelbuild/bazel#17803]
### Description of the bug:

It is extremely difficult to try to figure out how to add an `integrity` value to a Bzlmod override, and the logs are generally completely incorrect for suggestions.

I add an archive override:
```
archive_override(
    module_name = "gazelle",
    patches = [
        "//patches:bzlmod.patch",
    ],
    patch_strip = 1,
    strip_prefix = "bazel-gazelle-master",
    urls = ["https://github.com/bazelbuild/bazel-gazelle/archive/refs/heads/master.zip"],
)
```
The logs output:
```
DEBUG: Rule 'gazelle~override' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "a29f02861da70387de1193e17acdc9902aa3636ac3979a18e8d52289c23df676"
```
I try this, but get the error:
```
Error in archive_override: archive_override() got unexpected keyword argument 'sha256'
```
So, I then look up the `archive_override` fields, and add `integrity` field, to get the following errors:
```
Error in download_and_extract: Checksum error in repository @gazelle~override: Invalid SHA-256 SRI checksum 'sha256-a29f02861da70387de1193e17acdc9902aa3636ac3979a18e8d52289c23df676'
```

I try to go look on https://github.com/bazelbuild/bazel-central-registry/tree/main/modules to find the formatting of existing integrity values, but can't find anything.

So I go find an example: https://github.com/meteorcloudy/bzlmod-examples/blob/main/examples/06-archive-override/MODULE.bazel, and try to attach the `integrity = "sha256-zyeBn+s3YRUpFz1NjJ3WcZI9moLg7ImxZIkAksUNPJo="`, but then the logs show this:
```
Error in download_and_extract: java.io.IOException: Error downloading [https://github.com/bazelbuild/bazel-gazelle/archive/refs/heads/master.zip] to /home/user/.cache/bazel/_bazel_tfrench/b97476d719d716accead0f2d5b93104f/external/gazelle~override/temp15987837649025287268/master.zip: Checksum was a29f02861da70387de1193e17acdc9902aa3636ac3979a18e8d52289c23df676 but wanted cf27819feb37611529173d4d8c9dd671923d9a82e0ec89b164890092c50d3c9a
ERROR: Error computing the main repository mapping: unknown error during computation of main repo mapping
```
But this isn't even the value that I added to the rule.

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Add any incorrect `integrity` value to an override, and try to fix is using the error logs.

### Which operating system are you running Bazel on?

Linux

### What is the output of `bazel info release`?

release 6.0.0

### Have you found anything relevant by searching the web?

I was unable to find any documentation to help fix the issue.

**Comment by tyler-french** [comment]
It looks like this might be related to https://github.com/bazelbuild/bazel/issues/17123 and https://github.com/bazelbuild/bazel/issues/17124

I think this is more urgent than P2, since a mismatching/incorrect value is printed when the integrity is wrong.

**Comment by sgammon** [comment]
this has been helping us locally:

**in `~/bin/integrity`** (or wherever you prefer your scripts):
```bash
#!/bin/bash

#
# SRI hash generator utility from github.com/bazelbuild/bazel/issues/17803
#

set -eo pipefail

>&2 echo "Computing integrity for $1";
echo -n "sha256-";
curl -sSL -tlsv1.2 "$1" | openssl dgst -sha256 -binary | openssl base64 -A
```

**usage:**
```
$> integrity github.com/sgammon/rules_graalvm/archive/v0.9.0.tar.gz

Computing integrity for github.com/sgammon/rules_graalvm/archive/v0.9.0.tar.gz
sha256-ljI6wbeludsa4aOIxe0fuDDUYo06tLfwlThVgyHgMRE=
```

piping the output will yield the integrity hash only, so it works with `pbcopy` on a mac:
```
$> integrity github.com/sgammon/rules_graalvm/archive/v0.9.0.tar.gz | pbcopy
Computing integrity for github.com/sgammon/rules_graalvm/archive/v0.9.0.tar.gz

# On your clipboard:
# sha256-ljI6wbeludsa4aOIxe0fuDDUYo06tLfwlThVgyHgMRE=
```

**Comment by fmeum** [comment]
Should be fixed in 7.0.0 (except the doc part, which is covered by https://github.com/bazelbuild/bazel/issues/17124).

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.0.0 RC5](https://releases.bazel.build/7.0.0/rc5/index.html). Please test out the release candidate and report any issues as soon as possible. Thanks!
